### ISSUE TITLE
[F821] Improve error message for `anext`/`aiter`

### ISSUE THREAD

**Issue by JP-Ellis** [astral-sh/ruff#13287]
## Summary

The lint [F821](https://docs.astral.sh/ruff/rules/undefined-name/) has a false positive with `anext`.

As pointed out by @MichaReiser, this ultimately is due to having either `project.requires-python` or `tools.ruff.target-version` set to a too lower version.

In this instances, it was discovered that Ruff was targeting Python 3.8 (from an erroneous config).

## Example

```python
async def foo(iter: AsyncGenerator[str, None]):
    assert await anext(iter)
```

## Desired Behaviour

Where a symbol is introduced in a more recent version of Python, Ruff should:

1. Still raise F821 as it is strictly correct; but also,
2. Inform the use Ruff is assuming Python 3.x, and the symbol in question was only introduced in Python 3.y, and nudge the user to set `reures-python` or `target-version`.

## Environment

```console
❯ ruff --version
ruff 0.6.4

❯ python --version
Python 3.12.5
```

**Comment by evanarlian** [comment]
False positive on `aiter` as well.

**Comment by MichaReiser** [comment]
Have you specified ` requires-python` in your `pyproject.toml` or a `target-version` in your `ruff.toml`?  I'm asking because `anext` is only available in Python 3.10 and newer (and the default is Python 3.8)

**Comment by JP-Ellis** [comment]
Good catch. I have a monorepo and the root project has

```toml
[project]
# ...
requires-python = "~=3.12.0"

[tool.ruff]
target-version = "py312"
```

and one of the sub-packages has 

```toml
[project]
# ...
requires-python = "~=3.12.0"

[tool.ruff]
target-version = "py38"
```

I can confirm that changing from `py38` to `py312` fixes this.

In cases like there where there is a new symbol introduced in a more recent version of Python, should ruff instead suggest/direct the end-user to update the settings?

**Comment by MichaReiser** [comment]
> In cases like there where there is a new symbol introduced in a more recent version of Python, should ruff instead suggest/direct the end-user to update the settings?

That's a neat idea. I think the error message could possibly be improved by saying that the symbol is only defined in Python XX or newer.

**Comment by MichaReiser** [comment]
Thanks @AlexWaygood !
